data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlindedBlock.java

```diff
@@ -96,28 +108,77 @@ public PostBlindedBlock(final DataProvider dataProvider, final JsonProvider json
       })
   @Override
   public void handle(@NotNull final Context ctx) throws Exception {
-    try {
-      if (syncDataProvider.isSyncing()) {
-        ctx.status(SC_SERVICE_UNAVAILABLE);
-        ctx.json(BadRequest.serviceUnavailable(jsonProvider));
-        return;
-      }
+    adapt(ctx);
+  }
 
-      final SignedBeaconBlock signedBlindedBlock =
-          validatorDataProvider.parseBlindedBlock(jsonProvider, ctx.body());
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    if (syncDataProvider.isSyncing()) {
+      request.respondError(SC_SERVICE_UNAVAILABLE, "Beacon node is currently syncing.");
+    }
 
-      LOG.debug("parsed block is from slot: {}", signedBlindedBlock.getMessage().slot);
+    final SafeFuture<SendSignedBlockResult> result =
+        validatorDataProvider.submitSignedBlindedBlock(request.getRequestBody());
+    request.respondAsync(
+        result.thenApply(
+            blockResult -> {
+              if (blockResult.getRejectionReason().isEmpty()) {
+                return AsyncApiResponse.respondWithCode(SC_OK);
+              } else if (blockResult
+                  .getRejectionReason()
+                  .get()
+                  .equals(BlockImportResult.FailureReason.INTERNAL_ERROR.name())) {
+                return AsyncApiResponse.respondWithCode(SC_INTERNAL_SERVER_ERROR);
```

Comment:
Should we be including the error message here? `PostBlock` winds up including the rejection reason when reporting an error (empty response for SC_OK and SC_ACCEPTED).